src/JustSaying/AwsTools/MessageHandling/Dispatch/HandlerMap.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+using HandlerFunc = System.Func<JustSaying.Models.Message, System.Threading.Tasks.Task<bool>>;
+
+namespace JustSaying.AwsTools.MessageHandling.Dispatch
+{
+    /// <summary>
+    /// A <see cref="HandlerMap"/> is a register of handlers keyed by type and queue. Calling <see cref="Add"/>
+    /// with a queue name, type, and handler will cause the handler to be called when a message matching the type
+    /// arrives in the queue.
+    /// </summary>
+    public class HandlerMap
+    {
+        private readonly IMessageMonitor _messageMonitor;
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly Dictionary<(string queueName, Type type), HandlerFunc> _handlers
```

Comment:
I only mentioned this so that we can leave it open for us to add support for direct SNS handlers later without having to go back to the drawing board.